tests/api/test_sources_endpoint.py

```diff
@@ -0,0 +1,141 @@
+import json
+import pytest
+
+from share.models import Source, ShareUser
+
+
+PROPER_ICON_URL = 'https://staging-cdn.osf.io/preprints-assets/bitss/square_color_no_transparent.png'
+IMPROPER_ICON_URL = 'https://github.com/CenterForOpenScience/SHARE.git'
+INVALID_URL = 'invalidURL'
+
+
+def get_test_data(icon=PROPER_ICON_URL, home_page=None):
```

Comment:
I'm not sure how to make this a fixture without making a separate fixture for each url type which seems worse.